pkg/plugin/processor/builtin/impl/avro/encode.go

```diff
@@ -107,15 +71,33 @@ func parseEncodeConfig(ctx context.Context, c config.Config) (encodeConfig, erro
 type encodeProcessor struct {
 	sdk.UnimplementedProcessor
 
-	logger  log.CtxLogger
-	cfg     encodeConfig
-	encoder encoder
+	logger   log.CtxLogger
+	cfg      encodeConfig
+	encoder  encoder
+	registry schemaregistry.Registry
 }
 
 func NewEncodeProcessor(logger log.CtxLogger) sdk.Processor {
 	return &encodeProcessor{logger: logger}
 }
 
+func (p *encodeProcessor) MiddlewareOptions() []sdk.ProcessorMiddlewareOption {
+	return []sdk.ProcessorMiddlewareOption{
+		sdk.ProcessorWithSchemaDecodeConfig{
+			PayloadEnabled: lang.Ptr(false),
+			KeyEnabled:     lang.Ptr(false),
+		},
+		sdk.ProcessorWithSchemaEncodeConfig{
+			PayloadEnabled: lang.Ptr(false),
+			KeyEnabled:     lang.Ptr(false),
+		},
+	}
```

Comment:
As mentioned in slack, I think disabling this could lead to some unexpected behavior 😕